contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/estimator/MultiModalDrtLegEstimator.java

```diff
@@ -0,0 +1,74 @@
+package org.matsim.contrib.drt.extension.estimator;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.matsim.api.core.v01.population.Leg;
+import org.matsim.contrib.drt.extension.estimator.run.DrtEstimatorConfigGroup;
+import org.matsim.contrib.drt.extension.estimator.run.MultiModeDrtEstimatorConfigGroup;
+import org.matsim.contrib.drt.routing.DrtRoute;
+import org.matsim.contrib.dvrp.run.DvrpModes;
+import org.matsim.core.scoring.functions.ModeUtilityParameters;
+import org.matsim.core.utils.misc.OptionalTime;
+import org.matsim.modechoice.EstimatorContext;
+import org.matsim.modechoice.ModeAvailability;
+import org.matsim.modechoice.estimators.LegEstimator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+
+/**
+ * Aggregate class for informed-mode-choice that makes sure to invoke the correct estimator for each drt mode.
+ */
+public class MultiModalDrtLegEstimator implements LegEstimator<ModeAvailability> {
+
+	private static final Logger log = LogManager.getLogger(MultiModalDrtLegEstimator.class);
+
+	protected final Map<String, DrtEstimator> estimators = new HashMap<>();
+
+	@Inject
+	public MultiModalDrtLegEstimator(Injector injector) {
+
+		MultiModeDrtEstimatorConfigGroup config = injector.getInstance(MultiModeDrtEstimatorConfigGroup.class);
+
+		// Collect all mode specific bindings
+		for (DrtEstimatorConfigGroup el : config.getModalElements()) {
+
+			DrtEstimator instance = null;
+			try {
+				instance = injector.getInstance(Key.get(DrtEstimator.class, DvrpModes.mode(el.mode)));
+			} catch (Exception e) {
+				log.warn("Could not retrieve drt estimator for {}: {}", el.mode, e.getMessage());
+			}
+
+			estimators.put(el.mode, instance);
+		}
+	}
```

Comment:
I think you should use a map binder, i.e. `MapBinder<DvrpMode, DrtEstimator>` and bind estimators by mode in each `DrtEstimatorModule`.

Reply:
But the estimators use modal bindings and will need other modal components. Will it work when binding them as map only?

Reply:
I think you should be able to do something like this:
1. In each `DrtEstimatorModule.install()` call:
```java
MapBinder.newMapBinder(this.binder(), DvrpMode.class, DrtEstimator.class) .mapBinder.addBinding(DvrpModes.mode(getMode())).to(DrtEstimator.class);
```
2. Inject the map into the `MultiModalDrtLegEstimator` constructor:
```java
@Inject
	public MultiModalDrtLegEstimator(Map<DvrpMode, DrtEstimator> drtEstomators) {...}
```

This way you avoid handing DI (e.g. using injector) inside the "business logic" code.

BTW. I haven't tried this example with your code, because there is no unit test to play with, so it may not be 100% correct.

Reply:
Thanks, I got it working.